src/client/pages/OfferNew/Checkout/index.tsx

```diff
@@ -225,11 +220,12 @@ export const Checkout: React.FC<Props> = ({
       handleSignedEvent(member.data?.member ?? null)
     }
   }, [member.data?.member, signStatus?.signState, variation])
+
   useScrollLock(visibilityState, scrollWrapper)
 
   const canInitiateSign = Boolean(
-    signUiState !== SignUiState.STARTED &&
-      signUiState !== SignUiState.STARTED_WITH_REDIRECT &&
+    signUiState !== 'STARTED' &&
+      signUiState !== 'STARTED_WITH_REDIRECT' &&
```

Comment:
I really disagree actually - I think it's so much easier to read with this kind of string union type. I changed it because I found it so hard to read the code with the enum we previously had. But if everyone else likes enums more I can of course change it.

Reply:
I like unions as well 👌, but I'm guessing there's a reason that we have enums in most cases like this. Since I'm no typescript wiz, what's you take @palmenhq @virrios

Reply:
I asked @palmenhq about that before, like more generally the difference between using enums and unions like these, and the way I understood it from his answers and from how enums works in these kinds of cases, they are completely interchangeable.

Reply:
Fun tidbit since, we did this last week, (thought it was not necessary once we removed all danish type of Contracts and got removed)
Enums are iterable.  It was not at first possible for types but was later introduced, with this syntax
`
const contracts = ['travel', 'home', 'accident'] **as const;**
type IterableContractsType = typeof contracts[number]; // 'travel' | 'home' | 'accident'
`
I don't mind the enum, in terms of readability, but now its not really typesafe since its a numeric enum? And its actually passed as props in some components. 
I don't mind the readability of the union type either, but comparing when you are using it as to compare states I feel like its more natural to have it as a enum. 

I am not that strongly opinionated in this though 😄

Reply:
> I don't mind the enum, in terms of readability, but now its not really typesafe since its a numeric enum? And its actually passed as props in some components.

I'm not sure I understand fully @virrios - what is not really typesafe?

But yes, it's true that enums are iterable but in this case we're not using that (or any other of the special properties of the enum, as far as I can see) so here it's perfectly possible to use a union with strings. What I like is the readability when referencing those values, which in my view comes from the enum reference being more cluttered:
```javascript
  // less clutter :)
  signUiState !== 'STARTED'

  // more clutter
  signUiState !== SignUiState.STARTED
```
Especially confusing in these particular files where everything is called something with "sign" and "state" or "status" 😄

Reply:
> I'm not sure I understand fully @virrios - what is not really typesafe?
> 
> But yes, it's true that enums are iterable but in this case we're not using that (or any other of the special properties of the enum, as far as I can see) so here it's perfectly possible to use a union with strings. What I like is the readability when referencing those values, which in my view comes from the enum reference being more cluttered:

Sry what I meant regarding that the enums are iterable was just to reference last week when we discussed the **strange** `as const` or `<const>['....']` syntax and I thought it was interesting and **fun** information **but not really applicable to what we are doing here 😄** 

**Regarding typesafeness** 
**This is numeric Enum because its auto incremented** 
`
enum Example {
a,
b
}
`
Its not type safe in the sense that If you pass it to a function, you can pass an arbitrary number as well. This type of enum is useful to use as a mask. However,  I saw that we use it as a property in another component and I think that you can pass just a number and it would not cause an error?. **This is my understanding** of numeric enums and  **correct me if I am wrong here!**.

So the union type would be more typesafe (or make the enum a string enum). 
So this was more to add to the general discussion.

** Regarding readability ***
I like enums and I think they are readable and not necessarily more cluttered.


Maybe we should decide when/when not to use enums/union? I like what you said in the meeting, so it looks like one person is writing the code.

Reply:
@virrios Ah, now I see! Interesting, didn't really know that about enums. But yes, let's have that discussion elsewhere then, and leave this with the changes I made since those are more type safe as Victor said?

Reply:
Sorry i'm a bit late to the party here... Interesting point about the enums vs numbers, i had no idea but as it seems typescript is happy with the following code? Seems bad to me. I guess it's quite obvious that you shouldn't interchange enums with numbers but you really have a point here. For this case i think union types is good.

```ts
enum MyEnum { A, B }
const fn = (blah: MyEnum) => blah

fn(MyEnum.A) // OK
fn(0) // OK
fn(12) // OK??? like wat
```